lib/latte/Install.py

```diff
@@ -119,30 +108,27 @@
     txt = subprocess.check_output(cmd,stderr=subprocess.STDOUT,shell=True)
     print(txt.decode('UTF-8'))
   except subprocess.CalledProcessError as e:
-    print("Make failed with:\n %s" % e.output.decode('UTF-8'))
-    sys.exit(1)
+    sys.exit("Make failed with:\n %s" % e.output.decode('UTF-8'))
 
 # create 3 links in lib/latte to LATTE dirs
-# do this -b or -p is set
-
-if buildflag or pathflag:
-  print("Creating links to LATTE files")
-  if os.path.isfile("includelink") or os.path.islink("includelink"):
-    os.remove("includelink")
-  if os.path.isfile("liblink") or os.path.islink("liblink"):
-    os.remove("liblink")
-  if os.path.isfile("filelink.o") or os.path.islink("filelink.o"):
-    os.remove("filelink.o")
-  cmd = 'ln -s "%s/src" includelink' % lattedir
-  subprocess.check_output(cmd,stderr=subprocess.STDOUT,shell=True)
-  cmd = 'ln -s "%s" liblink' % lattedir
-  subprocess.check_output(cmd,stderr=subprocess.STDOUT,shell=True)
-  cmd = 'ln -s "%s/src/latte_c_bind.o" filelink.o' % lattedir
-  subprocess.check_output(cmd,stderr=subprocess.STDOUT,shell=True)
+
+print("Creating links to LATTE files")
+if os.path.isfile("includelink") or os.path.islink("includelink"):
+  os.remove("includelink")
+if os.path.isfile("liblink") or os.path.islink("liblink"):
+  os.remove("liblink")
+if os.path.isfile("filelink.o") or os.path.islink("filelink.o"):
+  os.remove("filelink.o")
+cmd = 'ln -s "%s/src" includelink' % lattedir
```

Comment:
```python
os.symlink(os.path.join(lattedir, 'src'), 'includelink')
os.symlink(lattedir, 'liblink')
os.symlink(os.path.join(lattedir, 'src', 'latte_c_bind.o'), 'filelink.o')
```

Reply:
Thanks. This is a whole new chunk of changes, but since the overall purpose is to increase portability and remove dependencies on external tools, `os.path.join()` and `os.symlink()` should be used. ...and while at it, the `tarfile` module can be used to extract the tar archives, too. Back to the emacs session...